mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/PrepareGameScreenState.kt

```diff
@@ -1,23 +1,34 @@
 package ch.epfl.sdp.mobile.ui.prepare_game
 
-/**
- * State interface of the [PrepareGameScreen]
- * @property colorChoice chosen color
- * @property gameType chosen game type
- */
-interface PrepareGameScreenState {
+import ch.epfl.sdp.mobile.ui.social.Person
+
+/** State interface of the [PrepareGameScreen] */
+interface PrepareGameScreenState<P : Person> {
+
+  /** Color choices for a chess game */
+  enum class ColorChoice {
+    White,
+    Black
+  }
+
+  /** The chosen color for the authenticated user */
   var colorChoice: ColorChoice
-  var gameType: GameType
-}
 
-/** Color choices for a chess game */
-enum class ColorChoice {
-  White,
-  Black
-}
+  /** The list of opponents to display in the [PrepareGameScreen] */
+  val opponents: List<P>
+
+  /** A potentially selected opponent to display differently in the opponent list */
+  var selectedOpponent: P?
+
+  /** The action to take when clicking on an opponent in the opponent's list */
+  fun onOpponentClick(opponent: P)
+
+  /**
+   * A callback for the action to take when clicking on the "play" button in the dialog, when a
+   * specific opponent is selected.
+   */
+  fun onPlayClick(opponent: P)
```

Comment:
This should not pass the opponent and it should have the signature `fun onPlayClick()` instead.

Reply:
Fixed in [3a5c8ed](https://github.com/epfl-SDP/android/pull/200/commits/3a5c8ed6ec211afc80268cf3f0b69bdadbeabdfd)